### ISSUE TITLE
Hyperparameter tuning

### ISSUE THREAD

**Issue by lkksharma** [rohitinu6/Stock-Price-Prediction#10]
In most of these algorithms, the basic parameters are taken to be default. I wish to add hyper-parameter tuning using Grid Search CV in SVR and KNN to see if the error reduces and Random Forest for experimental purposes.

I am a GGSoC Ext '24 Contributor. Kindly assign this issue to me for fixing.

Also, kindly add (gssoc-ext) and (hacktoberfest-accepted) as labels
